getnacelle/nacelle-react
# [packages/react-hooks] Persist Checkout ID and Checkout Complete Status in Local Storage

### Stories

- [DD-233](https://nacelle.atlassian.net/browse/DD-233) 

> In @nacelle/react-hooks, persist checkoutId and checkoutComplete

- [DD-258](https://nacelle.atlassian.net/browse/DD-258)

> In @nacelle/react-hooks, throw an error if the useCheckout hook isn't supplied all of the requisite credentials


### Type of Change


### What is being changed and why?

#### What

- When using the `useCart`'s `cartActions.setCheckoutStatus` to set the `cartState.checkoutId` and `cartState.checkoutComplete`, persist the values in `window.localStorage`, so that they can be used to re-set those values in memory when returning to the site from a Shopify checkout.
- The `useCheckout` hook now throws an error if it's not supplied complete `credentials`

#### Why

- When merchant developers were using `cartActions.setCheckoutStatus` to set the `checkoutId` and `checkoutComplete` values returned from the `useCheckout` hook's `checkoutData.data.processCheckout` payload, those values would be lost after navigating to a Shopify checkout.
- Merchant developers could inadvertently use the `useCheckout` hook without supplying the necessary `credentials`



### Demonstration

https://user-images.githubusercontent.com/5732000/115094862-17488b00-9eed-11eb-93f8-fb44e84029a8.mov

### How to Test



- check out the feature branch
- `cd examples/nextjs && npm run dev`
- navigate to `localhost:4000`
- add items to cart, perform a checkout, then return to `localhost:4000`
- use the [React dev tools](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en) to verify the presence of `checkoutId` and `checkoutComplete`, as shown in the above demonstration video
